src/main/java/org/vanilladb/core/storage/index/btree/BTreeIndex.java

```diff
@@ -254,6 +256,20 @@ private void search(SearchRange searchRange, SearchPurpose purpose) {
 		leaf = new BTreeLeaf(dataFileName, leafblk, keyType, searchRange, tx);
 	}
 
+	private boolean isFileEmpty(String fileName) {
+		// Optimization
+		// Assume we won't delete the BtreeIndex.
+		// Once the index file is not empty, the file won't be empty again
+		
+		boolean cacheMiss = fileEmptyCache.get(fileName) == null;
+		// Call fileSize(fileName) again, if cache miss or the cache says the file is empty.
+		if (cacheMiss || fileEmptyCache.get(fileName)) {
+			fileEmptyCache.put(fileName, fileSize(fileName) == 0);
+		}
+		
+		return fileEmptyCache.get(fileName);

```

Comment:
There is a problem here. If it finds a file is empty, it would always think the file is empty even after it appends a new block. This is because it does not update the cache after it adds new blocks.

Reply:
Actually, if it finds a file is empty, it would always recheck whether this file is still empty until a block is appended.

 If the file is empty, fileEmptyCache.get(fileName) returns True.
```
if (cacheMiss || fileEmptyCache.get(fileName)) {
    fileEmptyCache.put(fileName, fileSize(fileName) == 0);
}
```

Reply:
Oh, yes! You're right! I missed that part. Thx!